backend/inspirehep/snow/api.py

```diff
@@ -0,0 +1,571 @@
+    def _update_ticket_with_inspire_recid(self, ticket_id, recid, assignee):
+        """Update ticket with inspire recid
+        Creates a new, third party ticket and write its id to the original ticket
+        Args:
+            ticket_id (str): ticket id
+            recid (str): record recid
+            assignee (str): id of the user who should be added to the ticket as an assigned person
+        """
+        found_third_party_ticket_id = self._get_third_party_tickets_for_recid(recid)
+        if found_third_party_ticket_id:
+            third_party_ticket_id = found_third_party_ticket_id[0]["sys_id"]
+        else:
+            third_party_ticket_payload = {
+                "u_third_party_ticket_id": recid,
+            }
+            third_party_ticket_id = self.create_ticket(
+                endpoint=self.third_party_ticket_endpoint,
+                payload=third_party_ticket_payload,
+            )
+
+        ticket_update_payload = {
+            "assigned_to": assignee or self.inspire_user_id,
+            "u_current_task_state": self.ticket_status_mapping["waiting"],
+            "u_third_party": "INSPIRE",
+            "u_third_party_ticket": third_party_ticket_id,
+        }
+        self.edit_ticket(ticket_id, payload=ticket_update_payload)
+        if not assignee:
+            self.edit_ticket(ticket_id, payload={"assigned_to": ""})
+
+    def _get_third_party_tickets_for_recid(self, recid):
+        """Returns all tickets that are associated with the given recid.
+
+        Args:
+            recid (int): The record id.
+        """
+        search_query = f"u_third_party_ticket_id={recid}"
+        ticket = self.search_snow(self.third_party_ticket_endpoint, search_query)
+        return ticket
+
+    def _prepare_query_string_from_parameters_dict(self, query_string_parameters):
+        """Helper to prepare request query string in a SNOW specific way
+        Args:
+            query_string_parameters (dict): query string parameters
+        Returns:
+            query_string (str): formatted query string
+        """
+        query_string = "^".join(
+            f"{key}={value}" for key, value in query_string_parameters.items()
+        )
+        return query_string
+
+    def get_tickets_by_recid(self, recid, exclude_resolved=True):
+        """Returns all tickets that are associated with the given recid.
+
+        Args:
+            recid (int): The record id.
+            exclude_resolved (boolean): should resolved tickets be excluded from the response
+        Returns:
+            tickets_for_recid (list): list of found tickets
+        """
+        third_party_search_query_string = f"u_third_party_ticket_id={recid}"
+        tickets_search_parameters = dict(u_third_party="INSPIRE")
+        if exclude_resolved:
+            tickets_search_parameters[
+                "u_current_task_state"
+            ] = f"{self.ticket_status_mapping['waiting']}^ORu_current_task_state={self.ticket_status_mapping['assigned']}"
```

Comment:
```suggestion
            ] = f"{self.ticket_status_mapping['waiting']}^ORu_current_task_state={self.ticket_status_mapping['assigned']}"
```

👆 is `^OR` correct?

Reply:
yes, unfortunately to add or condition we need to link it with `^OR`, from docs: [`^OR: Logically OR multiple query statements.`](https://docs.servicenow.com/bundle/tokyo-application-development/page/integrate/inbound-rest/concept/c_RESTAPI.html)